src/cryptsetup/cryptsetup.c

```diff
@@ -1322,8 +1322,11 @@ static int attach_luks_or_plain_or_bitlk_by_tpm2(
                                 return log_error_errno(SYNTHETIC_ERRNO(EAGAIN), "TPM2 PIN unlock failed, falling back to traditional unlocking.");
                         if (ERRNO_IS_NOT_SUPPORTED(r)) /* TPM2 support not compiled in? */
                                 return log_debug_errno(SYNTHETIC_ERRNO(EAGAIN), "TPM2 support not available, falling back to traditional unlocking.");
-                        if (r != -EAGAIN) /* EAGAIN means: no tpm2 chip found */
-                                return r;
+                        /* EAGAIN means: no tpm2 chip found */
+                        if (r != -EAGAIN) {
+                                log_debug_errno(r, "TPM2 operation failed, falling back to traditional unlocking: %m");
```

Comment:
hmm, so I think we should upgrade this to `log_notice_errno()`. This log line carries information that the error logging done by acquire_tpm2_key() is not giving users, i.e. that we now fall back to traditional unlocking. Thus I think this should be above LOG_DEBUG log level, so that users can actually see this

Reply:
ok, done